File: c_rust_day_3/postgres_format_specifier.c
#include<stdio.h>

int
main() {

    // Integer format specifier
    short int  si = 2; // 2 byte int
    printf("%hd\n", si);

    int i = 4; // 4 byte int
    printf ("%d\n", i);

    long int li = 8; // 8 byte int
    printf("%ld\n", li);

    // Float format specifier
    float f = 4; // 4 byte fractional number
    printf("%f\n", f); 
    
    double d = 8; // 8 byter fractional number
    printf("%lf\n", d);

    //Character format specifier
    char c = '1'; // 1 byte character
    printf("%c\n", c);

    // String format specifier
    // There is no type `string` in C.
    // But, there is format specifer for as we have for primitive types.
    char s[333] = {'1', '1'}; // Represents the string '11' which is char array.
    printf("%s\n", s);

    return 0;
}

File: c_rust_day_3/postgres_format_specifier.rs
fn main() {

    //Integer format specifiers
    let _i8: i8 = 1; // 1 byte
    let _i16: i16 = 2; // 2 byte
    let _i32: i32 = 4; // 4 byte
    let _i64: i64 = 8; // 8 byte
    let _i128: i128 = 16; // 16 byte

    println!("{}{}{}{}{}", _i8, _i16, _i32, _i64, _i128);

    //Float format specifiers
    let _f32: f32 = 4.0; // 4 bytes
    let _f64: f64 = 8.0; // 8 bytes

    println!("{}{}", _f32, _f64);

    //Character format specifiers
    let _char: char = '1'; // 4 bytes

    println!("{}", _char);

    //Bool format specifier
    let _bool: bool = true; // 1 byte

    println!("{}", _bool);

}
